src/_pytest/pytester.py

```diff
@@ -332,10 +332,15 @@ def countoutcomes(self) -> List[int]:
         return [len(x) for x in self.listoutcomes()]
 
     def assertoutcome(self, passed: int = 0, skipped: int = 0, failed: int = 0) -> None:
-        realpassed, realskipped, realfailed = self.listoutcomes()
-        assert passed == len(realpassed)
-        assert skipped == len(realskipped)
-        assert failed == len(realfailed)
+        __tracebackhide__ = True
+
+        outcomes = self.listoutcomes()
+        realpassed, realskipped, realfailed = outcomes
+        assert {"passed": passed, "skipped": skipped, "failed": failed} == {
```

Comment:
Through the tests we usually write "assert obtained == expected" (for example `assert f() == 4`), but here the comparison is swapped... for consistency I think this should be swapped as well.

```python
expected = {"passed": passed, "skipped": skipped, "failed": failed}
obtained = {
    "passed": len(realpassed),
    "skipped": len(realskipped),
    "failed": len(realfailed),
}
assert obtained == expected
```

Reply:
Amended.